### DISCUSSION TITLE
"Wait time before cure" suggestion is not applying 'first empty layer' on Tango file

### DISCUSSION THREAD

**Discussion by bryanseger** [sn4k3/UVtools#720] [General]
So I set up a Mars 3 in Voxeldance Tango, slicing it into a .goo file, opening that up in UVtools and applying my usual "wait time before cure" setting. It is applying the appropriate wait times as expected, but it is not creating that first 'blank dummy' layer that I am expecting. This works fine going from Lychee to UVtools, but not from Tango to UVtools. 

I have attached both the .goo file directly from Tango and the .ctb file that I saved out from UVtools. Am I missing something here, or is it something in Tango that's not playing nice with UVtools?

I like that first dummy layer as the first true layer gets the expected wait-before-cure time. Perhaps there's another method for creating that first dummy layer?
[Desktop.zip](https://github.com/sn4k3/UVtools/files/11666478/Desktop.zip)

___

Before the "wait time before cure" suggestion was added to UVtools, there was some sort of "add layer" or "insert layer" type button up here that I would use to add a fake first dummy layer, but that option seems to be no more, or it got moved and I cannot find it. Is that functionality still present somewhere?

![image](https://github.com/sn4k3/UVtools/assets/4185392/e095113c-8cb0-4ff1-bc05-e4b5097b9af6)

**Comment by sn4k3** [comment]
Read change-log of [3.14.0](https://github.com/sn4k3/UVtools/releases/tag/v3.14.0), in special that line:

> (Improvement) Suggestion - Wait time before cure: Create the empty layer only to file formats that we know who require it

As far I know, only CTB and derivate files have that annoyance of not respecting first layer settings, and so the empty layer is required for those but not for others. So, to cleanup and spare time it was improved that way.
You can inspect that first layer at print time to confirm if it's respecting the wait time and report back.

___

UVtools never had a button to add/insert blank layers as it is considered an issue by itself. It only have the clone function which is what you want to replicate the "empty layer" by cloning first layer and make it all black but add a fade pixel somewhere to not be flag as empty layer by the detection. It also can be added with scripting or the terminal but again, not required (I think)

**Reply by bryanseger** [reply]
> You can inspect that first layer at print time to confirm if it's respecting the wait time and report back.

I just made a file in Tango for the Mars 4 DLP, ran it through UVtools to give the first 3 layers a 45s wait time, and ran it on the printer. The 45s wait was not respected for the first layer, similar to what we've seen on .CTB files. So, I believe the .goo files have the same annoyance and would benefit from the "create a first empty layer" setting under the "wait time before cure" suggestion.

**Reply by sn4k3** [reply]
Strange that all follow ctb bad habits, rather than hack the format why not ask the devs to fix it? IMO it worth to inform them to see if they want to fix that too.

**Comment by bryanseger** [comment]
Ok I see that now. I have not tested the "first layer annoyance" yet on either Mars 4 Max or DLP (I've printed with both, just haven't paid attention to that detail yet). In this case, I was using Tango for the add-text feature that's built into it (which Lychee also has, but only for the Pro version), but I was trying to use Tango for a Mars 3 instead of 4, thus me using UVtools to convert it to a .ctb.

I did try taking the Tango .goo, using UVtools to convert it to a .ctb, and then reopening that .ctb inside of UVtools to see if I could then add the empty layer by using the 'wait before cure time' suggestion, but that did not work either.

I am able to use the clone function to create the empty layer by erasing the pixels essentially. I remember doing this many months ago, but I didn't have to erase the pixels. Did there used to be a 'create empty layer' option within the clone function?


Thanks for your quick responses by the way. My team greatly appreciates the level of detail and care you've put into this program and your fast response time.

**Reply by sn4k3** [reply]
I can confirm that CTB does create the empty layer if on version 3 or above. For Encrypted CTB it also creates.

> Did there used to be a 'create empty layer' option within the clone function?

It never was. I never allowed to create empty layers inside UI. In fact, this empty layer with a dummy pixel comes firstly after that suggestion come out.
You can blank out the layer with Pixel Arithmetic tool or do the whole thing with Terminal in like 3 lines.
